src/lib/support/TypeUtils.h

```diff
@@ -0,0 +1,38 @@
+/*
```

Comment:
Given that we already have other files containing custom cast functions like `SafeInt.h`, `CHIPSafeCasts.h`, `SafePointerCast.h` I think a name like `EnumCast.h` would be more precise in your case.

Reply:
My idea is putting some backports and other convenient templates that comes from or using C++ type_traits library in this file, not only the cast for enums.

Since this function lives in <utility> in C++ stl.